### PR TITLE
update test-new() function works with no module or stream

### PR DESCRIPTION
**Pull request by AarushiSingh09**
Hello!
I am an Outreachy Applicant can you please have a look at the update I have made.
Kindly guide me through this.
Thank you!

### CODE REVIEW

File: modulemd/v2/tests/test-modulemd-modulestream.c [hunk 1]
```diff
@@ -66,6 +66,24 @@ module_stream_test_construct (ModuleStreamFixture *fixture,
       g_assert_null (modulemd_module_stream_get_stream_name (stream));
 
       g_clear_object (&stream);
+    
+    
+      /*Test that the new() function works with no module or stream*/
+    
+      stream = modulemd_module_stream_new (version, "foo", "latest");
```
Comment (sgallagher) [inline modulemd/v2/tests/test-modulemd-modulestream.c]: This doesn't match the comment you made above it. You wrote "Test that the new() function works with no module or stream" and then you created `stream` with the module name "foo" and stream name "latest".

This is just a copy of the test at line 31.

Comment (sgallagher) [reply modulemd/v2/tests/test-modulemd-modulestream.c]: Also, the test at line 58 handles the case of no module or stream, so I'm not sure what this patch is meant to accomplish.

File: modulemd/v2/tests/test-modulemd-modulestream.c [hunk 2]
```diff
@@ -66,6 +66,24 @@ module_stream_test_construct (ModuleStreamFixture *fixture,
       g_assert_null (modulemd_module_stream_get_stream_name (stream));
 
       g_clear_object (&stream);
+    
+    
+      /*Test that the new() function works with no module or stream*/
```
Comment (sgallagher) [inline modulemd/v2/tests/test-modulemd-modulestream.c]: Please leave a space between the start and end of comments.
```suggestion
      /* Test that the new() function works with no module or stream */
```

File: modulemd/v2/tests/test-modulemd-modulestream.c [hunk 3]
```diff
@@ -66,6 +66,24 @@ module_stream_test_construct (ModuleStreamFixture *fixture,
...
+    
+      stream = modulemd_module_stream_new (version, "foo", "latest");
+      g_assert_nonnull (stream);
+      g_assert_true (MODULEMD_IS_MODULE_STREAM (stream));
+    
+      g_assert_cmpint (
+        modulemd_module_stream_get_mdversion (stream), ==, version);
+      g_assert_cmpstr (
+        modulemd_module_stream_get_module_name (stream), ==, "foo");
+      g_assert_null (modulemd_module_stream_get_module_name (stream));
```
Comment (sgallagher) [inline modulemd/v2/tests/test-modulemd-modulestream.c]: This will of course fail because you set the stream name to "foo" (and checked it just above). So this test will always fail.

Did you remember to run `ninja test` before submitting this patch?

File: modulemd/v2/tests/test-modulemd-modulestream.c [hunk 4]
```diff
@@ -66,6 +66,24 @@ module_stream_test_construct (ModuleStreamFixture *fixture,
...
+      g_assert_null (modulemd_module_stream_get_stream_name (stream));
```
Comment (sgallagher) [inline modulemd/v2/tests/test-modulemd-modulestream.c]: Same as above; you already tested that this matched the stream name. Why would you also test it for NULL? (And for that matter, if you meant to test for non-NULL, you should always do that before the comparison, since otherwise if it *is* NULL, the comparison will throw an exception because it's attempting to dereference a NULL pointer.

**Comment by AarushiSingh09** [comment]
Hello!
I might have misunderstood the code base and the requirements. I'll  again
go throught the code base and try to propose a better update.  Thank you
for the help.
I'll  ask you if I get stuck somewhere in between.
Thank you

**Comment by sgallagher** [comment]
@AarushiSingh09 Are you still working on this change? If not, please close this PR. If so, please provide an update.

**Comment by AarushiSingh09** [comment]
I tried to contribute but I think I am stuck terribly with the code base I
don't  know where to start from.  Can you please help?

**Comment by sgallagher** [comment]
@AarushiSingh09 I need more information to help you. What problem are you having? It might be best to join #fedora-modularity on Freenode IRC and I can help you more interactively.

**Comment by AarushiSingh09** [comment]
Hello sir!
Can you please send me the exact link to Join fedora-modularity on Freenode
IRC.

**Comment by sgallagher** [comment]
https://freenode.net/kb/answer/chat has basic information, then you need to register your nick: https://freenode.net/kb/answer/registration

Lastly, join the channel with `/join #fedora-modularity`

**Comment by AarushiSingh09** [comment]
hello!
I have tried this immense number of times now I am not receiving any
confirmation email from free node. I have tried it with so many Ids
different nicknames, even wrote to them. I am not able to join fedora
community. I really want to be a part of the organisation and all this is
making me worry more.

**Comment by OrionStar25** [comment]
Hi @AarushiSingh09 . I can help you with that if you want. Just send me an [email](chunnushrivastava@gmail.com) and I'll walk you through it :)
